packages/kubernetes-1.22/credential-provider-config-yaml

```diff
@@ -11,8 +11,15 @@ providers:
 {{/each}}
     defaultCacheDuration: "{{default "12h" this.cache-duration}}"
     apiVersion: credentialprovider.kubelet.k8s.io/v1alpha1
-{{#if (eq @key "ecr-credential-provider")}}
+{{#if (or (eq @key "ecr-credential-provider") this.env-variables)}}
     env:
+{{#if this.env-variables}}
+{{#each this.env-variables}}
+      - name: {{@key}}
+        value: {{this}}
```

Comment:
There are values accepted by `SingleLineString`, such as `"`, that would result in an invalid YAML file. We could either filter those out at the settings level, or else defensively quote them here.

```suggestion
        value: '{{this}}'
```


Rendering `settings.aws.profile` likely needs the same fix.